sched/Kconfig

```diff
@@ -44,6 +44,16 @@ config DISABLE_ENVIRON
 	bool "Disable environment variable support"
 	default DEFAULT_SMALL
 
+config DISABLE_SIGNALS
+	bool "Disable signal support"
+	default n
```

Comment:
let's change to DEFAULT_SMALL, like other DISABLE_XXX

Reply:
> Done. However, I think disabling signals is not only for reducing footprint, but also for improving performance and safety. 

yes, that's why you and we develop the patch to make the signal disable.

> When signals are enabled, a task’s stack may be reused to run a signal handler, which is not entirely safe.

the partial signal disable remove this path like the full disable, so this problem doesn't exist anymore with both approach.

> Moreover, various signal-related operations must be performed during task initialization, creation, context switching, and exit, all of which introduce additional performance overhead.

yes, that's why the full signal disable is useful, if the userspace application doesn't use any signal related functionality.